### ISSUE TITLE
No support for xenserver_redirect_to_master

### ISSUE THREAD

**Issue by zozoh94** [fog/fog-xenserver#72]
Foreman-xen project seems to use the argument xenserver_redirect_to_master with the compute class : 
`    def client
      @client ||= ::Fog::Compute.new(
        :provider                     => 'XenServer',
        :xenserver_url                => url,
        :xenserver_username           => user,
        :xenserver_password           => password,
        :xenserver_redirect_to_master => true
      )
    end`
but the code of your project seems to not support it.
Can you confirm that ? Did you consider to implement it ?

Thank you

**Comment by plribeiro3000** [comment]
Yeah, `redirect_to_master` is not supported.

The supported options are:

```ruby
      requires :xenserver_username
      requires :xenserver_password
      requires :xenserver_url
      recognizes :xenserver_defaults
      recognizes :xenserver_timeout
      recognizes :xenserver_use_ssl
      recognizes :xenserver_verify_mode
      recognizes :xenserver_port
```

See them [here](https://github.com/fog/fog-xenserver/blob/c1c6d244f51f1428453e588bb6a9e258f9a176f6/lib/fog/compute/xen_server.rb#L10-L17)

Looking back before extraction, it used to have support for it: https://github.com/fog/fog/blob/78e5a649602cfe1e783fe15f679589462c7cb1ab/lib/fog/xenserver/compute.rb#L14

But i could not confirm it was being used at all. It does seem it was a dead option thus it was removed.

Could you confirm if this options is something supported by `XenServer` or just a force redirect being made by `fog`?

**Comment by plribeiro3000** [comment]
Ok. I've found it: It used to do it here:

```ruby
          if @redirect_to_master == false
            @connection  = Fog::XenServer::Connection.new(@host, @timeout)
          elsif @redirect_to_master == true
            host_master  = @connection.find_pool_master(@username, @password)
            if host_master && host_master!= @host
              @host = host_master
              @connection  = Fog::XenServer::Connection.new(@host, @timeout)
            end
          end
```

This was added to `fog` but never to `fog-xenserver`. See fog/fog#3356.

Would you be able to open a PR adding it?

**Comment by plribeiro3000** [comment]
By the way, i'm kind of glad it never touched `fog-xenserver`. We can make it better now.
If i'm not mistaken, You can connect to a slave node but can't execute any commands so i guess whether we should always connect to the master node instead.

Yeah, the more i think, the more it seems wrong to have a configuration for it. If you connect to a slave node you wont be able to use fog at all so we should always force the connection to master.

WDYT?

**Comment by zozoh94** [comment]
Yes, I think too, the redirection should always be enabled to improve the resilience ! I will do a PR...

**Comment by plribeiro3000** [comment]
Awesome! Thank you @zozoh94 !
